### PR TITLE
Honoring Disk encryption set Id as input for VMWare to Azure and Hype…

### PR DESCRIPTION
**Pull request by TalluriAnusha**
…V to Azure replications

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by TalluriAnusha** [comment]
Optional input parameters are added for Azure Site Recovery Scenarios.

semantic errors and model validation error READONLY_PROPERTY_NOT_ALLOWED_IN_REQUEST, are being taken in other PRs.

**Comment by TalluriAnusha** [comment]
@MyronFanQiu , could you please review the PR?

**Comment by MyronFanQiu** [comment]
@TalluriAnusha Hello. Could you please solve CI? You should add examples for your new parameter.

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json [hunk 1]
```diff
@@ -8833,6 +8833,10 @@
           "description": "The VHD id.",
           "type": "string"
         },
+        "diskId": {
+          "description": "The disk id.",
```
[the remark this answers is not in the document]
Comment (TalluriAnusha) [reply specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json]: It will be a Guid.

**Comment by TalluriAnusha** [comment]
Examples were not updated as these new parameters are optional.

**Comment by TalluriAnusha** [comment]
@MyronFanQiu, can we have teams meeting? I couldn't find your alias to setup.

**Comment by MyronFanQiu** [comment]
@TalluriAnusha Of course. Just wonder what's the objective of the meeting?

**Comment by MyronFanQiu** [comment]
@raych1 Hello. Can you take a look at this RP? @TalluriAnusha doesn't want to fix these CI errors in this PR. We may need admin right to merge it if we decide to do that.

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (pilor) [reply specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json]: @TalluriAnusha Does the user provide this or is it generated by the service?   Improve the description so users know what this is.  "ID" can mean far too many things.  By default I would think this was a disk's resource ID.

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json [hunk 2]
```diff
@@ -10468,6 +10476,13 @@
         "useManagedDisks": {
           "description": "A value indicating whether managed disks should be used during failover.",
           "type": "string"
+        },
+        "diskIdToDiskEncryptionMap": {
+          "description": "The disk Id to disk encryption set map.",
```
Comment (pilor) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json]: Again, please provide additional description around what "disk ID" is and what its looking for in the value (disk encryption set RESOURCE ids)

**Review by pilor — CHANGES_REQUESTED** [review]

**Comment by raych1** [comment]
@TalluriAnusha , for semantic failures about discriminator, you could suppress it this time. Here's the example suppression code to be added to `readme.md`
``` yaml
directive:
  - from: notRequiredDiscriminatorWithSuppression.json
    suppress: OAV131 
    reason: 
```
The model validation failures were there before, please create task in your backlog to fix the examples, otherwise this failures will still appears in next PR.

**Comment by TalluriAnusha** [comment]
@raych1, thank you. readme.md is updated through other PR https://github.com/Azure/azure-rest-api-specs/pull/7982

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json [hunk 3]
```diff
@@ -10468,6 +10476,13 @@
         "useManagedDisks": {
           "description": "A value indicating whether managed disks should be used during failover.",
           "type": "string"
+        },
+        "diskIdToDiskEncryptionMap": {
+          "description": "The dictionary of disk resource Id to disk encryption set ARM Id.",
```
Comment (pilor) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json]: The use of "resource id" or "ARM id" is inconsistent throughout but not part of your change.  However, having a description that uses both doesn't make sense.  "resource ID" is my preferred terminology but at the least it should be consistent.

**Review by pilor — APPROVED** [review]
One suggested description change

**Comment by TalluriAnusha** [comment]
Semantic error DISCRIMINATOR_NOT_REQUIRED and Model validation error OBJECT_MISSING_REQUIRED_PROPERTY will be addressed through separate PR.

Added backlog item to update the examples.

**Comment by raych1** [comment]
> Semantic error DISCRIMINATOR_NOT_REQUIRED and Model validation error OBJECT_MISSING_REQUIRED_PROPERTY will be addressed through separate PR.
> 
> Added backlog item to update the examples.

As synced with @TalluriAnusha offline, we force merged this PR because the failures are not related to the changes in this PR and the failures are tracked to be fixed.

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json (hunk 3 above)
Comment (TalluriAnusha) [reply specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json]: sure, will revisit the descriptions.

**Pull request merged**